### ISSUE TITLE
Remove `{CameraDescription/Optics/Telescope}.from_name` methods

### ISSUE THREAD

**Issue by maxnoe** [cta-observatory/ctapipe#1978]
**Please describe the use case that requires this feature.**

While these are handy shortcuts for tests etc. it turned out to also create a lot of bugs and surprising behavior,
e.g. people analyzing / plotting data simulated with Prod5 using the `from_name` version, which loads an older geometry for most telescopes.

**Describe the solution you'd like**

We should remove these methods and make it clear that the subarray information should always be read from the input file or some other source instead of relying on the `from_name` mechanism.

**Describe alternatives you've considered**

Making the `from_name` system more complex by including more information like versions and prod numbers. But I don't like it, we should just remove the method instead of making the leaky abstraction more complex.

**Comment by kosack** [comment]
The original intent was that we would have a selectable set of input data for this and other tests, e.g. one per prod.  See #738 for example.  It just never got implemented.

**Comment by maxnoe** [comment]
@kosack I think we can better enable this through fixtures, e.g. in the pull request #1985 I introduce fixtures like `subarray_prod5_paranal`.

**Comment by maxnoe** [comment]
There was an issue again: #2157 we should probably go ahead with this...

**Comment by STSpencer** [comment]
So it's been a while since I commented on anything with respect to ctapipe, but since this has an effect on a number of ongoing projects with current generation instruments I'd like to make a request. I agree that getting rid of the from_name functionality makes sense as you propose @maxnoe; I think the big problem is that many users don't understand the different camera model versions. It's also not currently possible (as far as I can tell) to determine the available options for this method (i.e. what files are on the data server) or to tell where the associated camera geometry files actually came from.

Going to a purely from_url/from ctapipe-process file-type system is good (because it's more difficult to abuse it). But somewhere in the documentation I'd ask there be downloadable links to historical camera geometry files, who contributed them, and what exact instrument/simulation model they correspond to. The reality is that we're going to be dealing with (i.e.) CHEC-S on ASTRI data or H.E.S.S. data not in ctapipe-compatible data formats for some time yet.

**Comment by maxnoe** [comment]
> The reality is that we're going to be dealing with (i.e.) CHEC-S on ASTRI data or H.E.S.S. data not in ctapipe-compatible data formats for some time yet.

Just to comment on this part for now: that involves IO plugins and other custom code by the Experte for those systems. These IO plugins should provide the corresponding instrument descriptions by whatever means they deem appropriate and should not rely on the ctapipe test data server to provide them.

I.e. `ctapipe_io_lst` bundles the latest, real LST 1 geometry and single pe pulse shape files.

**Comment by kosack** [comment]
> @kosack I think we can better enable this through fixtures, e.g. in the pull request https://github.com/cta-observatory/ctapipe/pull/1985 I introduce fixtures like subarray_prod5_paranal.

Yes but the current fixtures are a bit of a mess, and I don't see a nice way to iterate through a set of them. E.g. I would like to be able to have test functions like:

```py3
@pytest.parametrize(dataset=['prod3','prod5b', 'LST-1'])
@pytest.parametrize(camera=['LSTCam','NectarCam'])
def test_something(dataset, camera):
     utils.set_dataset(dataset)
     geom = CameraGeometry.from_name(camera)
     ...
```

To test a function across all implementations of a particular instrument info.
